scrapy/core/http2/stream.py

```diff
@@ -178,47 +220,69 @@ def receive_data(self, data: bytes, flow_controlled_length: int):
 
     def receive_headers(self, headers):
         for name, value in headers:
-            self._response_headers[name] = value
-
-    def reset(self):
-        """Received a RST_STREAM -- forcefully reset"""
-        # TODO:
-        #  Q1. Do we need to send the request again?
-        #  Q2. What response should we send now?
-        self.stream_closed_server = True
-        self._cb_close(self.stream_id)
+            self._response['headers'][name] = value
 
-    def lost_connection(self):
-        # TODO: Same as self.reset
-        self.stream_closed_server = True
-        self._cb_close(self.stream_id)
+    def close(self, event=None):
+        """Based on the event sent we will handle each case.
 
-    def end_stream(self):
-        """Stream is ended by the server hence no further
+        event: StreamEnded
+        Stream is ended by the server hence no further
         data or headers should be expected on this stream.
-
         We will call the response deferred callback passing
         the response object
+
+        event: StreamReset
+        Stream reset via RST_FRAME by the upstream hence forcefully close
+        this stream and send TODO: ?
+
+        event: None
+        No event is launched -- Hence we will simply close this stream
         """
+        # TODO: In case of abruptly stream close
+        #  Q1. Do we need to send the request again?
+        #  Q2. What response should we send now?
```

Comment:
None, I think. It probably makes sense to call the request errback at this point. See https://github.com/scrapy/scrapy/blob/master/scrapy/core/downloader/handlers/http11.py#L542